platform/openide.filesystems/src/org/netbeans/modules/openide/filesystems/declmime/FileElement.java

```diff
@@ -205,6 +221,18 @@ private void readExternal(DataInput in) throws IOException {
                     names.add(new FileName(in));
                 }
             }
+            int fileNamePatternsSize = -1;
+            try {
+                fileNamePatternsSize = in.readInt();
```

Comment:
I'd like to see a test covering this part - e.g. to read old format as well as new format.